src/Compilers/CSharp/Portable/Binder/Semantics/Operators/BinaryOperatorOverloadResolution.cs

```diff
@@ -809,41 +807,69 @@ private static LiftingResult UserDefinedBinaryOperatorCanBeLifted(TypeSymbol lef
             // SPEC: better than all other function members, then the function member invocation is ambiguous and a binding-time 
             // SPEC: error occurs.
 
-            // UNDONE: This is a naive quadratic algorithm; there is a linear algorithm that works. Consider using it.
+            int currentBestIndex;
             var candidates = result.Results;
-            for (int i = 1; i < candidates.Count; ++i)
+            for (currentBestIndex = 0; currentBestIndex < candidates.Count; currentBestIndex++)
+            {
+                if (candidates[currentBestIndex].Kind == OperatorAnalysisResultKind.Applicable)
+                {
+                    break;
+                }
+            }
+
+            if (currentBestIndex == candidates.Count)
+            {
+                return;
```

Comment:
> Once we reach here there is no candidate in candidates array.

Thanks for pointing this out. You are correct, there a no applicable candidates and we can return early.